singer_sdk/helpers/_schema.py

```diff
@@ -0,0 +1,136 @@
+# pylint: disable=redefined-builtin, too-many-arguments, invalid-name
+"""Provides an object model for JSON Schema."""
+
+import json
+
+from singer import Schema
+
+# These are keys defined in the JSON Schema spec that do not themselves contain
+# schemas (or lists of schemas)
+STANDARD_KEYS = [
+    "title",
+    "description",
+    "minimum",
+    "maximum",
+    "exclusiveMinimum",
+    "exclusiveMaximum",
+    "multipleOf",
+    "maxLength",
+    "minLength",
+    "format",
+    "type",
+    "required",
+    "enum",
+    # These are NOT simple keys (they can contain schemas themselves). We could
+    # consider adding extra handling to them.
+    "additionalProperties",
+    "anyOf",
+    "patternProperties",
+]
+
+
+class SchemaPlus(Schema):  # pylint: disable=too-many-instance-attributes
```

Comment:
@Jack-Burnett wdyt of making this a `dataclass`, similar to `singer_sdk.helpers._singer.CatalogEntry`?

That'd would save some redundant code in the constructor and give us a nice `__repr__` and `__eq__` for free.

Reply:
Yeah that sounds like a good idea, done.
It also meant I had to give some type hints which are nice too, though I left a couple at Any for the more flexible json-schema fields